### PR TITLE
Implement UI for logic rule descriptions

### PR DESCRIPTION
**Pull request by sergei-maertens**
Related to #2450 - depends on #2524

* Allows explicitly providing your own rule description
* Backend description generation for when it's not explicitly provided
* Supports the `_meta.description` extension to simplify nested parts

https://watch.screencastify.com/v/ry6S1ymjGOArH1IycFoi

### CODE REVIEW

File: src/openapi.yaml [hunk 1]
```diff
@@ -2863,6 +2863,112 @@ paths:
...
+
+        This endpoint takes into account (optionally) client-provided sub-expression
+        descriptions via the ``_meta.description`` path. E.g. an input like:
+
+        ```json
+        {
+          "==": [
+            {"var": "foo"},
+            {
+                "+": [{"var": "a", "var": "b"}],
```
Comment (SilviaAmAm) [inline src/openapi.yaml:2880]: Shouldn't this be `[{"var": "a"}, {"var": "b"}]`?

File: src/openforms/js/components/admin/form_design/logic/LogicDescription.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  const onChange = event => {
+    const {
+      target: {value},
+    } = event;
+    _onChange(event);
+    setModifiedByHuman(value !== '');
+  };
+
+  const mustBail = isEmpty(logicExpression) || !generationAllowed || modifiedByHuman || hasFocus;
```
Comment (SilviaAmAm) [inline src/openforms/js/components/admin/form_design/logic/LogicDescription.js:43]: Maybe if the expression is invalid JSON it should also bail?

File: src/openapi.yaml (hunk 1 above)
Comment (sergei-maertens) [reply src/openapi.yaml:2880]: yes, nice catch!

File: src/openforms/js/components/admin/form_design/logic/LogicDescription.js (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/logic/LogicDescription.js:43]: this should already the case as the `logicExpression` prop needs to be valid JSON - and it's only updated from the JSONWidget to the parent state when the JSON validates. But it's possible something broke there while I was trying to clean up the code. Is this a theoretical remark or did you see it happen? :P

Comment (SilviaAmAm) [reply src/openforms/js/components/admin/form_design/logic/LogicDescription.js:43]: It was theoretical! :smile:

**Pull request merged**